src/main/java/org/myrobotlab/service/interfaces/ImagePublisher.java

```diff
@@ -0,0 +1,33 @@
+package org.myrobotlab.service.interfaces;
+
+import org.myrobotlab.framework.interfaces.NameProvider;
+import org.myrobotlab.service.data.ImageData;
+
+public interface ImagePublisher extends NameProvider {
+  
+  // These are all the methods that the image publisher should produce.
+  public static String[] publishMethods = new String[] {"publishImage"};
+  
+  /**
+   * Publishes an image
+   * @param data - standard java.net.URI local files are
+   * file:///path 
+   */
+ public ImageData publishImage(ImageData data);
+ 
+ default public void attachImageListener(ImageListener display) {
+   attachImageListener(display.getName());
+ }
+ 
+ // Default way to attach an image listener so implementing classes need 
+ // not worry about these details.
+ default public void attachImageListener(String name) {
```

Comment:
for completeness, please add detach/remove ImageListener.

Reply:
good call - updated

Reply:
I don't see the removeImageListener method added...

Reply:
correct - because removeImageListener isn't standard
detachImageListener is